core/qbft/qbft.go

```diff
@@ -695,7 +717,7 @@ func filterRoundChange[I any, V Value[V]](msgs []Msg[I, V], round int64) []Msg[I
 func filterMsgs[I any, V Value[V]](msgs []Msg[I, V], typ MsgType, round int64, value *V, pr *int64, pv *V) []Msg[I, V] {
 	var (
 		resp []Msg[I, V]
-		dups = make(map[dedupKey]bool)
```

Comment:
we already did this in `filterMsgs` but since we do not always use this function, generalise the `uniqSource[I, V]()` helper function